pkg/cli/cmd/credential/register/azure/sp/serviceprincipal.go

```diff
@@ -0,0 +1,191 @@
+in order to create or manage resources contained in the group. The resource group should be created before
+calling 'rad credential register azure sp'.
+` + common.LongDescriptionBlurb,
+		Example: `
+# Register (Add or update) cloud provider credential for Azure with service principal authentication
+rad credential register azure sp --client-id <client id/app id> --client-secret <client secret/password> --tenant-id <tenant id>
+`,
+		Args: cobra.ExactArgs(0),
+		RunE: framework.RunCommand(runner),
+	}
+
+	commonflags.AddOutputFlag(cmd)
+	commonflags.AddWorkspaceFlag(cmd)
+
+	cmd.Flags().String("client-id", "", "The client id or app id of an Azure service principal.")
+	_ = cmd.MarkFlagRequired("client-id")
+
+	cmd.Flags().String("client-secret", "", "The client secret or password of an Azure service principal.")
+	_ = cmd.MarkFlagRequired("client-secret")
+
+	cmd.Flags().String("tenant-id", "", "The tenant id of an Azure service principal.")
+	_ = cmd.MarkFlagRequired("tenant-id")
+
+	return cmd, runner
+}
+
+// Runner is the runner implementation for the `rad credential register azure sp` command.
+type Runner struct {
+	ConfigHolder      *framework.ConfigHolder
+	ConnectionFactory connections.Factory
+	Output            output.Interface
+	Format            string
+	Workspace         *workspaces.Workspace
+
+	ClientID     string
+	ClientSecret string
+	TenantID     string
+	KubeContext  string
+}
+
+// NewRunner creates a new instance of the `rad credential register azure sp` runner.
+func NewRunner(factory framework.Factory) *Runner {
+	return &Runner{
+		ConfigHolder:      factory.GetConfigHolder(),
+		ConnectionFactory: factory.GetConnectionFactory(),
+		Output:            factory.GetOutput(),
+	}
+}
+
+// Validate runs validation for the `rad credential register azure sp` command.
+//
+
+// Validate checks for the presence of a workspace, output format, client ID, client secret and tenant ID, and
+// sets them in the Runner struct if they are present. If any of these are not present, an error is returned.
+func (r *Runner) Validate(cmd *cobra.Command, args []string) error {
+	workspace, err := cli.RequireWorkspace(cmd, r.ConfigHolder.Config, r.ConfigHolder.DirectoryConfig)
+	if err != nil {
+		return err
+	}
+	r.Workspace = workspace
+
+	format, err := cli.RequireOutput(cmd)
+	if err != nil {
+		return err
+	}
+	r.Format = format
+
+	clientID, err := cmd.Flags().GetString("client-id")
+	if err != nil {
+		return err
+	}
+
+	clientSecret, err := cmd.Flags().GetString("client-secret")
+	if err != nil {
+		return err
+	}
+
+	tenantID, err := cmd.Flags().GetString("tenant-id")
+	if err != nil {
+		return err
+	}
+
+	r.ClientID = clientID
+	r.ClientSecret = clientSecret
+	r.TenantID = tenantID
+
+	kubeContext, ok := r.Workspace.KubernetesContext()
+	if !ok {
+		return clierrors.Message("A Kubernetes connection is required.")
+	}
+	r.KubeContext = kubeContext
```

Comment:
Do you know why the kubecontext is required here? it looks to be unused.

Reply:
looks like it was unused. removed